### PR TITLE
Add more detailed instructions for releases

### PR DESCRIPTION
**Pull request by brancz**
While doing the v2.11 release I didn't quite remember some details, so I wanted to make sure I captured them in our release docs, so here it is.

@brian-brazil @beorn7 @simonpasquier @krasi-georgiev @codesome @gouthamve @juliusv

### CODE REVIEW

**Comment by brian-brazil** [comment]
:+1:

**Comment by krasi-georgiev** [comment]
:+1:

**Comment by codesome** [comment]
:+1:

**Comment by juliusv** [comment]
:+1:

File: RELEASE.md [hunk 1]
```diff
@@ -88,9 +88,9 @@ Now all you can do is to wait for tarballs to be uploaded to the Github release
 
 If the release has happened in the latest release branch, merge the changes into master.
 
-To update the docs, a PR needs to be created to `prometheus/docs`. See [this PR](https://github.com/prometheus/docs/pull/952/files) for inspiration.
+To update the docs, a PR needs to be created to `prometheus/docs`. See [this PR](https://github.com/prometheus/docs/pull/952/files) for inspiration (note: only actually merge this for final releases, not for pre-releases like release-candidates).
```
Comment (beorn7) [inline RELEASE.md]: Nit: release-candidates → release candidate

**Pull request merged**